### PR TITLE
Align context menu order for FileStatusList

### PR DESCRIPTION
**Pull request by gerhardol**
## Proposed changes 

See the commits for detailed explanations, this is a summary of changes.

The context menus are not aligned much, neither contents nor order.
This PR tries to align the options in a x more consistent way, grouping similar functionality and aligning RevisionDiff, RevisionFile and FormCommit.
The order can be discussed, but I doubt I will update the screen shots.

The convention in menus is that the first item should be the default 'double-clicked' action and that it should be default.
This PR makes sure that items are bot bold if they are not the default and sets submodule "Open with Git Extensions" as bold.
To change the default file action to open the history in RevDiff/RevFile, some deeper changes are required. If so, the History/Blame should probably be handled by FileStatusList as OpenWithGE is now. Just setting bold should be eenough, setting order will be a little complex -  I want DiffTool as default in RevDiff anyway

One functional change (not counting the setting to bold), Edit File was not blocked for FormCommit Edit file.

Some code removed too.

A squash is planned before merge, probably just one commit.

Note: This should be merged before 4.0, all screenshots in the documentation need to be updated anyway...

### Before
(Shortcut for Stage/Unstage from #7825, not in master)
pre-revdiff-file-worktree
pre-revdiff-subm-worktree

pre-revdiff-file-index
pre-revdiff-subm-index

pre-revdiff-file-commit
pre-revdiff-subm-commit

pre-revfile-subm
pre-revfile-file

pre-commit-file-worktree
pre-commit-subm-worktree

pre-commit-file-index
pre-commit-subm-index

Removed
pre-commit-subm-summary
pre-revdiff-subm-summary

### After

post-revdiff-file-worktree
post-revdiff-subm-worktree

post-revdiff-file-index
post-revdiff-subm-index

post-revdiff-file-commit
post-revdiff-subm-commit

post-revfile-file
post-revfile-subm

post-commit-file-worktree
post-commit-subm-worktree

post-commit-file-index
post-commit-subm-index

## Test methodology 
Manual

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/UserControls/FileStatusList.cs [hunk 1]
```diff
@@ -1146,18 +1151,18 @@ private void FileStatusListView_ContextMenu_Opening(object sender, CancelEventAr
...
 
-            if (isSubmoduleSelected)
+            if (isSubmoduleSelected && !DisableSubmoduleMenuItemBold)
             {
-                _openSubmoduleMenuItem.Font = AppSettings.OpenSubmoduleDiffInSeparateWindow ?
-                    new Font(_openSubmoduleMenuItem.Font, FontStyle.Bold) :
-                    new Font(_openSubmoduleMenuItem.Font, FontStyle.Regular);
+                _openSubmoduleMenuItem.Font = AppSettings.OpenSubmoduleDiffInSeparateWindow
+                    ? new Font(_openSubmoduleMenuItem.Font, FontStyle.Bold)
+                    : new Font(_openSubmoduleMenuItem.Font, FontStyle.Regular);
```
Comment (RussKie) [inline GitUI/UserControls/FileStatusList.cs:1165]: I'm not sure this is correct.

This needs to happen if a submodule is selected, irrespective of "bold or not".
The second part - i.e. set font weight - depends on two conditions now - `AppSettings.OpenSubmoduleDiffInSeparateWindow` and `DisableSubmoduleMenuItemBold`.

File: GitUI/UserControls/FileStatusList.cs [hunk 2]
```diff
@@ -56,6 +56,11 @@ public sealed partial class FileStatusList : GitModuleControl
         public new event EventHandler DoubleClick;
         public new event KeyEventHandler KeyDown;
         public new event EnterEventHandler Enter;
+
+        [Description("Disable showing open submodule menu items as bold")]
+        [DefaultValue(false)]
+        public bool DisableSubmoduleMenuItemBold { get; set; }
```
Comment (RussKie) [inline GitUI/UserControls/FileStatusList.cs:62]: A "bold" menu item implies "default" item that gets invoked on double click.
This property controls the visual aspect, i.e. item is bold or not, but I can't see where double-click action is enabled/disabled respectively.

Comment (RussKie) [inline GitUI/UserControls/FileStatusList.cs:62]: Maybe?
```suggestion
        public bool OpenSubmoduleAsDefaultAction { get; set; }
```

File: GitUI/UserControls/FileStatusList.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/UserControls/FileStatusList.cs:1165]: This was not correct before. See the screenshot for FormCommit Index 
image

This worked for FormCommit worktree as FormCommit created a separate menu item with the same name was created.

See below as well.

File: GitUI/UserControls/FileStatusList.cs (hunk 2 above)
Comment (gerhardol) [reply GitUI/UserControls/FileStatusList.cs:62]: Implementation is in each form using FileStatusList, not related to the menu item (that item is unknown in the forms).

Comment (gerhardol) [reply GitUI/UserControls/FileStatusList.cs:62]: This only controls the 'bold' part, not default action.
I also avoided setting this to true by default (I expect false to be default for the property).
(Note that the 'bold' is not set for the standard default action, File History.)

You are commenting on a core problem in the current implementation, which is valid but not something I tried to address here.
I primarily want to clean up the order for next release and fix that items were bold when the should not be.
Can we settle on the order and fixes to the current implementation in this PR, it has enough changes already?

Going forward, I would like to remove setting for OpenSubmoduleDiffInSeparateWindow (always enable, the name is misleading now) and use DiffTool as the default doubleclick. With this PR, the first item is also the default and more often what I believe is expected (except for RevFile, better to change for consistency?).
The _openSubmoduleMenuItem could be duplicated to all forms too as well as the bold handling, but that will duplicate a number of lines of code.

Comment (RussKie) [reply GitUI/UserControls/FileStatusList.cs:62]: Sounds sensible. Please add a "TODO" in the code.

Comment (gerhardol) [reply GitUI/UserControls/FileStatusList.cs:62]: Added a TODO, something should be added to the final commit message too (squash should probably be done, describing the shanges)

**Comment by RussKie** [comment]
Merge away

**Comment by gerhardol** [comment]
Any comments on the unified order of items? 
@mstv usually have something to say.

**Comment by RussKie** [comment]
Looks good to me :+1:
Awesome job btw

**Comment by gerhardol** [comment]
Squashed (switch order for SubmoduleUpdate and Commit), rebased
Long message...
This can be improved

**Comment by gerhardol** [comment]
@msftbot merge in 12 hours

**Comment by mstv** [comment]
Indended change? Also in `RevisionDiffControl.Designer.cs`?
grafik

**Comment by gerhardol** [comment]
>> Squashed (switch order for SubmoduleUpdate and Commit), rebased

> Indended change? Also in RevisionDiffControl.Designer.cs?

Yes, Update is likely more likely with the most common use of submodules and Commit is the least, as I see it.

But good catch if you missed my note, something better to compare rebase is needed #7909. In this case these files could be compared with the help of hopefully soon merged #7923 (even better with #7899)

**Pull request merged**